vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -942,114 +997,49 @@ def execute_model(
 
-        t0 = time.time()
-
-        # TODO temporary until 'pooler_output' makes it to a release version
-        # in vllm
-        extra_kwargs: dict[str, Any] = {}
-        if "pooler_output" in CBSpyreModelRunnerOutput.__dataclass_fields__:
-            extra_kwargs["pooler_output"] = None
-
-        self._update_states(scheduler_output)
-        # TODO: change to EMPTY_MODEL_RUNNER_OUTPUT, right now this
-        # will be a breaking change, or clumsy to make retrocompatible
-        # with conditional import
-        if not scheduler_output.total_num_scheduled_tokens:
-
-            # Return empty ModelRunnerOuptut if there's no work to do.
-            return CBSpyreModelRunnerOutput(req_ids=[],
-                                            req_id_to_index={},
-                                            sampled_token_ids=[],
-                                            spec_token_ids=None,
-                                            logprobs=None,
-                                            prompt_logprobs_dict={},
-                                            tkv=0,
-                                            **extra_kwargs)
-
-        model_input = self.prepare_model_input(scheduler_output)
-
-        # Marking dimensions static/dynamic
-        if model_input.is_prompt:
-
-            # batch static (batch size 1)
-            torch._dynamo.mark_static(model_input.input_tokens, 0)
-            torch._dynamo.mark_static(model_input.slot_mapping, 0)
-            torch._dynamo.mark_static(model_input.input_positions, 0)
-            torch._dynamo.mark_static(model_input.input_masks, 0)
-
-            # sequence dynamic
-            torch._dynamo.mark_dynamic(model_input.input_tokens, 1)
-            torch._dynamo.mark_dynamic(model_input.slot_mapping, 1)
-            torch._dynamo.mark_dynamic(model_input.input_positions, 1)
-            torch._dynamo.mark_dynamic(model_input.input_masks, 2)
-            torch._dynamo.mark_dynamic(model_input.input_masks, 3)
-
-        # decode
-        else:
-            # mask is no longer used here
-
-            # batch dynamic
-            torch._dynamo.mark_dynamic(model_input.input_tokens, 0)
-            torch._dynamo.mark_dynamic(model_input.block_table, 0)
-            torch._dynamo.mark_dynamic(model_input.slot_mapping, 0)
-            torch._dynamo.mark_dynamic(model_input.input_positions, 0)
-            torch._dynamo.mark_dynamic(model_input.current_tkv_mask, 0)
-            torch._dynamo.mark_dynamic(model_input.left_padded_prompt_mask, 0)
-
-            # sequence
-            torch._dynamo.mark_static(model_input.input_tokens, 1)  # always 1
-            torch._dynamo.mark_dynamic(model_input.block_table, 1)
-            torch._dynamo.mark_static(model_input.slot_mapping, 1)  # always 1
-            torch._dynamo.mark_static(model_input.input_positions,
-                                      1)  # always 1
+        output = self.base_execute_model(scheduler_output, **kwargs)
```

Comment:
Maybe I still haven't understood it fully, but could we not call `super().execute_model` here and rename the function in `class SpyreModelRunner` to `execute_model` ?

Reply:
Oh, okay, I have no idea about that. Let me read up on it.